### PR TITLE
BTAT-9691 - Util added that strips pound signs, commas and full stops

### PR DESCRIPTION
**Pull request by william-hatzar**
# Pull Request Checklist
* [ ] Branch is rebased with master
* [ ] Added Unit Tests for changed functionality
* [ ] Ran `sbt clean compile coverage test it:test coverageReport`, all tests pass and coverage meets minimum
* [ ] Checked libraries are up to date with `sbt validate`. (If applicable) library updates are done in a separate, single commit
* [ ] Ran [submit-vat-return-acceptance-tests](https://github.com/hmrc/submit-vat-return-acceptance-tests) (see README of repo)
* [ ] Ran [vat-returns-performance-tests](https://github.com/hmrc/vat-returns-performance-tests) (see README of repo)
* [ ] (If applicable) raised app-config-* PRs
* [ ] (If applicable) ran Wave test for changes to views
* [ ] (If applicable) added Integration Tests

### CODE REVIEW

File: app/utils/StripCharUtil.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+object StripCharUtil {
+
+  def stripAll(text: String, charactersToRemove: String): String = {
```
Comment (lewishay) [inline app/utils/StripCharUtil.scala:21]: I find this quite confusing to read. The recursive functions inside aren't using the clearest argument names (`a`, `n`) and recursive functions in general aren't the easiest to follow. Is this over-engineered a little? As per my comment below, if you can strip out `£`, `,` and ` ` with a single line, what is this function doing? Purely removing full stops?

The guidance suggests to only allow full stops to indicate pence. So the only case there is one full stop at the end. Could you just use `stripSuffix` at the end of the validation to remove a potential full stop from the end of the value?

File: app/utils/StripCharUtil.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        else start(1 + n)
+
+      @scala.annotation.tailrec def end(a: Int, n: Int): String =
+        if (n <= a) text.substring(a, n)
+        else if (charactersToRemove.indexOf(text.charAt(n - 1)) < 0) text.substring(a, n)
+        else end(a, n - 1)
+
+      val result = start(0)
+
+    result.replaceAll(",","")
```
Comment (lewishay) [inline app/utils/StripCharUtil.scala:35]: I'm confused why `£` and `,` are being passed into this util as the `charactersToRemove` string, but the logic in this single line removes all of them anyway.

File: app/utils/StripCharUtil.scala (hunk 1 above)
Comment (lewishay) [reply app/utils/StripCharUtil.scala:21]: Can we remove need `charactersToRemove` argument seeing as everywhere this function is called, the same value is being passed in?

File: test/utils/StripCharUtilSpec.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import uk.gov.hmrc.play.test.UnitSpec
+import utils.StripCharUtil.stripAll
+
+class StripCharUtilSpec extends UnitSpec {
```
Comment (lewishay) [inline test/utils/StripCharUtilSpec.scala:22]: One likely scenario that would be good to add a test for is a user entering a full stop as pence, e.g. `£10.56.`

Comment (lewishay) [reply test/utils/StripCharUtilSpec.scala:22]: Can you add tests to prove removal of leading/trailing whitespace.

File: test/utils/StripCharUtilSpec.scala [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    stripAll("£100,,,","£ ,") shouldBe "100"
+  }
+  "return the expected string if pound signs, commas and a full stop is present" in {
+    stripAll("£100.56,,,...","£ , .") shouldBe "100.56"
+  }
+  "return the expected string if a pound sign is included in the middle and full stops and commas are present" in {
+    stripAll("£100£.56...,,","£ . ,") shouldBe "100.56"
+  }
+  "return the expected string when commas are present" in {
+    stripAll("£100,000,000","£") shouldBe "100000000"
```
Comment (lewishay) [inline test/utils/StripCharUtilSpec.scala:37]: The fact you did not pass in a `,` to your function but it still stripped commas out from the value further proves that as mentioned above there is no need for that second parameter.

**Review by lewishay — CHANGES_REQUESTED** [review]

File: test/utils/StripCharUtilSpec.scala (hunk 3 above)
Comment (lewishay) [reply test/utils/StripCharUtilSpec.scala:22]: Can you add in a unit test for a standard value that doesn't need any stripping e.g. 100.56

**Review by lewishay — CHANGES_REQUESTED** [review]
